blog/_src/posts/2017-09-27-tutorial-contributing-to-racket.md

```diff
@@ -0,0 +1,414 @@
+Note that many of these tests are written using a custom test library.
+Thankfully there are plenty of nearby examples showing how to use this library.
+
+Implementation files may define a `test` submodule.
+Try grepping for `module.*test`
+
+> If you are not sure where to put new tests,
+>  then just add a comment to your pull request with a test program.
+> A repo manager can tell you where the test belongs.
+
+
+### Finding the docs
+
+Most large package repos have a directory named `<PKG>-doc/` that contains the
+ documentation.
+Look for a directory named `scribblings/` under the `-doc/` directory.
+
+Small packages usually have a top-level directory named `scribblings/` with documentation.
+At any rate, the package's `info.rkt` file should have a `scribbings` field
+ whose value contains a path to the documentation.
+
+Documentation for core Racket is in the [`RKT_DIR/pkgs/racket-doc/scribblings/`](https://github.com/racket/racket/tree/master/pkgs/racket-doc/scribblings)
+ directory.
+In particular, the Racket Reference starts [here](https://github.com/racket/racket/blob/master/pkgs/racket-doc/scribblings/reference/reference.scrbl).
+
+If you can find the HTML version of the documentation using `my-raco docs`,
+  the filename at the end of the URL might match the name of a `.scrbl` file
+  in the package.
+
+
+## 3. Edit the code, add tests and documentation
+
+First, create a branch for your changes.
+
+```
+$ git checkout -b my-branch
+```
+<br/>
+
+Otherwise, this step is completely straightforward :).
+Commit often, and don't forget the documentation and tests.
+
+
+## 4. Test the changes
+
+Always compile and run the files you edit. This is important even for Scribble files.
+
+To recompile the package and build documentation, run:
+
+```
+$ my-raco setup <PKG>
+```
+<br/>
+
+> `my-raco setup` will compile all packages and rebuild all documentation
+
+"How to run the unit tests" varies between packages ...
+
+
+### Running test submodules
+
+For files with a `test` submodule, run:
+
+```
+$ my-raco test <FILE.rkt>
+```
+
+DrRacket should run the test submodule whenever you click "Run".
+If it doesn't, check "Choose Language" `->` "Show Details" `->` "Submodules to Run".
+
+
+For files with a "test" submodule that uses a different name, e.g. `stress-tests`, run:
+
+```
+$ my-raco test -s stress-tests <FILE.rkt>
+```
+<br/>
+
+In DrRacket, you can change the "Submodules to Run" setting.
+
+
+### Running package-wide tests
+
+For many packages, the following command runs all tests:
+
+```
+$ raco test -c <PKG>
+```
+<br/>
+
+Other repos have special commands for running tests.
+The [`racket`](https://github.com/racket/racket)
+ and [`typed-racket`](https://github.com/racket/typed-racket)
+ are two examples.
+For these, look at the scripts `.travis.yml` and/or `appveyor.yml` to find
+ the exact commands to run.
+
+These CI tests will run when you push the pull request, but it's good to check
+ before you push.
+
+
+### 5) Create a PR
```

Comment:
FWIW, it's possible to make tiny cosmetic changes — say, fixing typos in the docs — by editing the file directly on the Github web site and making a PR right from there. The cheaper a PR is to make, the more likely it is to be made …